app/models/document_capture_session.rb

```diff
@@ -16,7 +16,7 @@ def store_result_from_response(doc_auth_response)
       id: generate_result_id,
     )
     session_result.success = doc_auth_response.success?
-    session_result.pii = doc_auth_response.pii_from_doc
+    session_result.pii = doc_auth_response.pii_from_doc.to_h
```

Comment:
More for my own learning - can you explain why the `to_h` is necessary?

Reply:
This is getting written to redis shortly after this. Downstream it is serialized into JSON so this doesn't _really_ matter now but I'm doing it to be defensive against a couple issues:

1. If the storage layer didn't serialize this into JSON then there might be a case where it tries to marshal the `Pii::StateId` struct which could lead to unpredictable results
2. Related to above, if it is written as a `Pii::StateId` struct successfully then there is a risk it gets demarshaled as the same which the code in the controller that unwraps this is not prepared to handle
3. The TrueID client still returns a hash at this point so this ensures consistent results between the two clients. This is especially important since this is the one we use for testing and the TrueID one is only running in upper environments.

Since the struct is just a key/value pair then converting to a hash and then back into the struct on read saves us some potential trouble since the serialization and storage layer may change underneath us and it helps ensure consistency with the other client.

Reply:
TIL [marshalling](https://en.wikipedia.org/wiki/Marshalling_(computer_science))! Thank you for that explanation, it really helped me understand.